src/server/handlers/dynamic.js

```diff
@@ -103,90 +46,33 @@ export default ({ server, config }) => {
       // Match Routes
       // this should only have one route as we force "exact" on each route
       // How would we error out if two routes match here? "Ambigous routes detected?" maybe earlier in app
-      const { route, match } = matchRoutes(routes, request.url.pathname)
+      const { route } = matchRoutes(routes, request.url.pathname)
 
       log.debug(`Matched route ${chalk.green(route.path)}`)
-      // This needs tidying
-      // If there's a branch of the route config, we have a route
-      // Optimistic default component data for static routes
-      let componentData = {
-        status: 200,
-        message: '200'
-      }
 
-      // Set a flag for whether we have a missing component later on
-      const routeComponentUndefined = typeof route.component === 'undefined'
-      // If we have an endpoint
-      if (route.endpoint) {
-        // Start to try and fetch data
-        const multidata = await fetchFromEndpointConfig({
-          endpointConfig: route.endpoint,
-          baseUrl: baseUrlResolver(config, request.url),
-          requestUrlObject: request.url,
-          params: match.params
-        })
-        componentData = normalizeApiResponse(multidata, route)
-      }
-      if (componentData.code > 299 || routeComponentUndefined) {
-        log.debug(`Render Error component`, {
-          componentData,
-          routeComponentUndefined
-        })
+      const requestPath = currentPath
+      const requestQuery = queryParams
+      const { responseString, status } = await tapestryRender(
```

Comment:
All of the heavy lifting required to render a page is now abstracted behind `tapestryRender`, and it only requires `tapestry.config.js`, `requestPath` and `requestQuery` as an input.

In the future it'd be easy to swap out Hapi for another framework - and it makes it possible for us to compile a tapestryRender lambda function - excluding the rest of the Tapestry framework, and Hapi from that bundle.